client/src/Main.purs

```diff
@@ -214,33 +215,32 @@ setupEditor { code } = do
     compile
 
   JQuery.select "#gist_save" >>= JQuery.on "click" \e _ ->
-    publishNewGist
+    launchAff_ publishNewGist
 
   compile
   cacheCurrentCode
 
 loadFromGist
   :: String
-  -> ({ code :: String } -> Effect Unit)
-  -> Effect Unit
-loadFromGist id_ k = do
-  runContT (runExceptT (getGistById id_ >>= \gi -> tryLoadFileFromGist gi "Main.purs")) $
-    case _ of
-      Left err -> do
-        window >>= alert err
-        k { code: "" }
-      Right code -> k { code }
+  -> ({ code :: String } -> Aff Unit)
+  -> Aff Unit
```

Comment:
Did you consider changing this function to have the type `String -> Aff { code :: String }` or perhaps even `String -> Aff String`? I think callback ought not to be necessary any more now that we're working in Aff; the benefit of Aff is that it lets us not bother with callbacks, so I think we might as well take advantage of that?

Reply:
I can go ahead and update these so they aren’t using callbacks anymore.